monkey/infection_monkey/exploit/HostExploiter.py

```diff
@@ -124,3 +127,39 @@ def add_executed_cmd(self, cmd):
         """
         powershell = True if "powershell" in cmd.lower() else False
         self.exploit_info["executed_cmds"].append({"cmd": cmd, "powershell": powershell})
+
+    def publish_propagation_event(
+        self,
+        target: str,
+        propagation_success: bool,
+        exploiter_name: str = "",
+        tags: frozenset = frozenset(),
+        error_message: str = "",
+    ):
+        propagation_event = PropagationEvent(
+            source=get_agent_id(),
+            target=IPv4Address(target),
+            success=propagation_success,
+            exploiter_name=exploiter_name or self.__class__.__name__,
```

Comment:
I'd say just use `self.__class__.__name__` for now. Or `self._EXPLOITED_SERVICE`.